Source/WebCore/layout/formattingContexts/inline/InlineFormattingContext.cpp

```diff
@@ -67,8 +67,8 @@ static std::optional<InlineItemRange> partialRangeForDamage(const InlineItemList
         ASSERT_NOT_REACHED();
         return { };
     }
-    auto& damagedInlineItem = inlineItemList[damageStartPosition.index];
-    if (damageStartPosition.offset && (!is<InlineTextItem>(damagedInlineItem) || damageStartPosition.offset >= downcast<InlineTextItem>(damagedInlineItem).length())) {
+    auto* damagedInlineItem = dynamicDowncast<InlineTextItem>(inlineItemList[damageStartPosition.index]);
+    if (damageStartPosition.offset && (!damagedInlineItem || damageStartPosition.offset >= damagedInlineItem->length())) {
```

Comment:
I think this change makes the code less readable. `is<InlineTextItem>` is not strictly a prerequisite, but it expresses the fact that non-text type inline items cannot have offsets.